## Black branes in Lifshitz-like spacetimes

This Jupyter/SageMath worksheet implements some computations of the article
- I. Ya. Aref'eva, A. A. Golubtsova & E. Gourgoulhon: *Analytic black branes in 
  Lifshitz-like backgrounds and thermalization*,
  [arXiv:1601.06046](http://arxiv.org/abs/1601.06046)

These computations are based on [SageManifolds](http://sagemanifolds.obspm.fr) (v0.9)

The worksheet file (ipynb format) can be downloaded from [here](https://github.com/sagemanifolds/SageManifolds/raw/master/Worksheets/v0.9/SM_black_brane.ipynb).

First we set up the notebook to display mathematical objects using LaTeX formatting:

In [1]:
%display latex

### Space manifold

Let us declare $M$ as a 5-dimensional manifold:

In [2]:
M = Manifold(5, 'M')
print(M)

5-dimensional differentiable manifold M


We introduce a coordinate system on $M$:

In [3]:
X.<t,x,y1,y2,r> = M.chart('t x y1:y_1 y2:y_2 r')
X

Chart (M, (t, x, y1, y2, r))

Next, we define the metric tensor, which depends on some real number $\nu$ and some arbitary function $f$:

In [4]:
g = M.lorentzian_metric('g')
var('nu', latex_name=r'\nu', domain='real')
ff = function('f')(r)
g[0,0] = -ff*exp(2*nu*r)
g[1,1] = exp(2*nu*r)
g[2,2] = exp(2*r)
g[3,3] = exp(2*r)
g[4,4] = 1/ff
g.display()

g = -e^(2*nu*r)*f(r) dt*dt + e^(2*nu*r) dx*dx + e^(2*r) dy1*dy1 + e^(2*r) dy2*dy2 + 1/f(r) dr*dr

If $f(r)=1$, this is the metric of a Lifshitz spacetime; if, in addition $\nu=1$, $(M,g)$ is a Poincaré patch of $\mathrm{AdS}_5$.

### Curvature

The Riemann tensor is

In [5]:
Riem = g.riemann()
print(Riem)

Tensor field Riem(g) of type (1,3) on the 5-dimensional differentiable manifold M


In [6]:
Riem.display_comp(only_nonredundant=True)

Riem(g)^t_x,t,x = -nu^2*e^(2*nu*r)*f(r) - 1/2*nu*e^(2*nu*r)*d(f)/dr 
Riem(g)^t_y1,t,y1 = -nu*e^(2*r)*f(r) - 1/2*e^(2*r)*d(f)/dr 
Riem(g)^t_y2,t,y2 = -nu*e^(2*r)*f(r) - 1/2*e^(2*r)*d(f)/dr 
Riem(g)^t_r,t,r = -1/2*(2*nu^2*f(r) + 3*nu*d(f)/dr + d^2(f)/dr^2)/f(r) 
Riem(g)^x_t,t,x = -nu^2*e^(2*nu*r)*f(r)^2 - 1/2*nu*e^(2*nu*r)*f(r)*d(f)/dr 
Riem(g)^x_y1,x,y1 = -nu*e^(2*r)*f(r) 
Riem(g)^x_y2,x,y2 = -nu*e^(2*r)*f(r) 
Riem(g)^x_r,x,r = -1/2*(2*nu^2*f(r) + nu*d(f)/dr)/f(r) 
Riem(g)^y1_t,t,y1 = -nu*e^(2*nu*r)*f(r)^2 - 1/2*e^(2*nu*r)*f(r)*d(f)/dr 
Riem(g)^y1_x,x,y1 = nu*e^(2*nu*r)*f(r) 
Riem(g)^y1_y2,y1,y2 = -e^(2*r)*f(r) 
Riem(g)^y1_r,y1,r = -1/2*(2*f(r) + d(f)/dr)/f(r) 
Riem(g)^y2_t,t,y2 = -nu*e^(2*nu*r)*f(r)^2 - 1/2*e^(2*nu*r)*f(r)*d(f)/dr 
Riem(g)^y2_x,x,y2 = nu*e^(2*nu*r)*f(r) 
Riem(g)^y2_y1,y1,y2 = e^(2*r)*f(r) 
Riem(g)^y2_r,y2,r = -1/2*(2*f(r) + d(f)/dr)/f(r) 
Riem(g)^r_t,t,r = -nu^2*e^(2*nu*r)*f(r)^2 - 3/2*nu*e^(2*nu*r)*f(r)*d(f)/dr - 1/2*e^(2*nu*r)*f(r)*d^2(f)/dr^2 
Riem(g)^r_x,x,r = nu^2*e^(2*nu*r)*f(r) + 1/2*nu*e^(2*nu*r)*d(f)/dr 
Riem(g)^r_y1,y1,r = e^(2*r)*f(r) + 1/2*e^(2*r)*d(f)/dr 
Riem(g)^r_y2,y2,r = e^(2*r)*f(r) + 1/2*e^(2*r)*d(f)/dr

The Ricci tensor:

In [7]:
Ric = g.ricci()
print(Ric)

Field of symmetric bilinear forms Ric(g) on the 5-dimensional differentiable manifold M


In [8]:
Ric.display()

Ric(g) = (2*(nu^2 + nu)*e^(2*nu*r)*f(r)^2 + (2*nu + 1)*e^(2*nu*r)*f(r)*d(f)/dr + 1/2*e^(2*nu*r)*f(r)*d^2(f)/dr^2) dt*dt + (-2*(nu^2 + nu)*e^(2*nu*r)*f(r) - nu*e^(2*nu*r)*d(f)/dr) dx*dx + (-2*(nu + 1)*e^(2*r)*f(r) - e^(2*r)*d(f)/dr) dy1*dy1 + (-2*(nu + 1)*e^(2*r)*f(r) - e^(2*r)*d(f)/dr) dy2*dy2 - 1/2*(4*(nu^2 + 1)*f(r) + 2*(2*nu + 1)*d(f)/dr + d^2(f)/dr^2)/f(r) dr*dr

In [9]:
Ric.display_comp()

Ric(g)_t,t = 2*(nu^2 + nu)*e^(2*nu*r)*f(r)^2 + (2*nu + 1)*e^(2*nu*r)*f(r)*d(f)/dr + 1/2*e^(2*nu*r)*f(r)*d^2(f)/dr^2 
Ric(g)_x,x = -2*(nu^2 + nu)*e^(2*nu*r)*f(r) - nu*e^(2*nu*r)*d(f)/dr 
Ric(g)_y1,y1 = -2*(nu + 1)*e^(2*r)*f(r) - e^(2*r)*d(f)/dr 
Ric(g)_y2,y2 = -2*(nu + 1)*e^(2*r)*f(r) - e^(2*r)*d(f)/dr 
Ric(g)_r,r = -1/2*(4*(nu^2 + 1)*f(r) + 2*(2*nu + 1)*d(f)/dr + d^2(f)/dr^2)/f(r)

The Ricci scalar:

In [10]:
Rscal = g.ricci_scalar()
print(Rscal)

Scalar field r(g) on the 5-dimensional differentiable manifold M


In [11]:
Rscal.display()

r(g): M --> R
   (t, x, y1, y2, r) |--> -2*(3*nu^2 + 4*nu + 3)*f(r) - (5*nu + 4)*d(f)/dr - d^2(f)/dr^2

### Source model
Let us consider a model based on the following action, involving a cosmological constant $\bar{\Lambda} = -\Lambda/2$ with $\Lambda>0$, a dilaton scalar field $\phi$ and a Maxwell 2-form $F$:

$$ S = \int \left( R(g) + \Lambda - \frac{1}{2} \nabla_m \phi \nabla^m \phi - \frac{1}{4} e^{\lambda\phi} F_{mn} F^{mn} \right) \sqrt{-g} \, \mathrm{d}^5 x \qquad\qquad \mbox{(1)}$$

where $R(g)$ is the Ricci scalar of metric $g$ and $\lambda$ is the dilatonic coupling constant.

### The dilaton scalar field

We consider the following ansatz for the dilaton scalar field $\phi$:
$$ \phi = \frac{1}{\lambda} \left( 4 r + \ln\mu \right) \iff e^{\lambda\phi} = \mu e^{4r},$$
where $\mu$ is a constant. 

In [12]:
var('mu', latex_name=r'\mu', domain='real')
var('lamb', latex_name=r'\lambda', domain='real')
phi = M.scalar_field({X: (4*r + ln(mu))/lamb}, 
                     name='phi', latex_name=r'\phi')
phi.display()

phi: M --> R
   (t, x, y1, y2, r) |--> (4*r + log(mu))/lamb

The 1-form $\mathrm{d}\phi$ is

In [13]:
dphi = phi.differential()
print(dphi)

1-form dphi on the 5-dimensional differentiable manifold M


In [14]:
dphi.display()

dphi = 4/lamb dr

In [15]:
dphi[:]  # all the components in the default frame

[0, 0, 0, 0, 4/lamb]

### The 2-form field

We consider the following ansatz for $F$:
$$ F = \frac{1}{2} q \, \mathrm{d}y_1\wedge \mathrm{d}y_2, $$
where $q$ is a constant. 
Let us first get the 1-forms $\mathrm{d}y_1$ and $\mathrm{d}y_2$:

In [16]:
X.coframe()

Coordinate coframe (M, (dt,dx,dy1,dy2,dr))

In [17]:
dy1 = X.coframe()[2]
dy2 = X.coframe()[3]

Then we can form $F$ according to the above ansatz:

In [18]:
var('q', domain='real')
F = q/2 * dy1.wedge(dy2)
F.set_name('F')
print(F)
F.display()

2-form F on the 5-dimensional differentiable manifold M


F = 1/2*q dy1/\dy2

By construction, the 2-form $F$ is closed (since $q$ is constant):

In [19]:
print(F.exterior_der())

3-form dF on the 5-dimensional differentiable manifold M


In [20]:
F.exterior_der().display()

dF = 0

Let us evaluate the square $F_{mn} F^{mn}$ of $F$:

In [21]:
Fu = F.up(g)
print(Fu)

Tensor field of type (2,0) on the 5-dimensional differentiable manifold M


In [22]:
F2 = F['_{mn}']*Fu['^{mn}']  # using LaTeX notations for contraction
print(F2)
F2.display()

Scalar field on the 5-dimensional differentiable manifold M


M --> R
(t, x, y1, y2, r) |--> 1/2*q^2*e^(-4*r)

We shall also need the tensor $\mathcal{F}_{mn} := F_{mp} F_n^{\ \, p}$:

In [23]:
FF = F['_mp'] * F.up(g,1)['^p_n']
print(FF)
FF.display()

Tensor field of type (0,2) on the 5-dimensional differentiable manifold M


1/4*q^2*e^(-2*r) dy1*dy1 + 1/4*q^2*e^(-2*r) dy2*dy2

The tensor field $\mathcal{F}$ is symmetric:

In [24]:
FF == FF.symmetrize()

True

Therefore, from now on, we set

In [25]:
FF = FF.symmetrize()

### Einstein equation

Let us first introduce the cosmological constant:

In [26]:
var('Lamb', latex_name=r'\Lambda', domain='real')

Lamb

From the action (1), the field equation for the metric $g$ is
$$ R_{mn} + \frac{\Lambda}{3} \, g  - \frac{1}{2}\partial_m\phi \partial_n\phi -\frac{1}{2} e^{\lambda\phi} F_{mp} F^{\ \, p}_n + \frac{1}{12} e^{\lambda\phi} F_{rs} F^{rs} \, g_{mn} = 0 $$
We write it as

    EE == 0

with `EE` defined by

In [27]:
EE = Ric + Lamb/3*g - 1/2* (dphi*dphi) - 1/2*exp(lamb*phi)*FF \
     + 1/12*exp(lamb*phi)*F2*g
EE.set_name('E')
print(EE)

Field of symmetric bilinear forms E on the 5-dimensional differentiable manifold M


In [28]:
EE.display_comp(only_nonredundant=True)

E_t,t = 2*(nu^2 + nu)*e^(2*nu*r)*f(r)^2 + (2*nu + 1)*e^(2*nu*r)*f(r)*d(f)/dr - 1/24*(mu*q^2 + 8*Lamb)*e^(2*nu*r)*f(r) + 1/2*e^(2*nu*r)*f(r)*d^2(f)/dr^2 
E_x,x = -2*(nu^2 + nu)*e^(2*nu*r)*f(r) - nu*e^(2*nu*r)*d(f)/dr + 1/24*(mu*q^2 + 8*Lamb)*e^(2*nu*r) 
E_y1,y1 = -2*(nu + 1)*e^(2*r)*f(r) - 1/12*(mu*q^2 - 4*Lamb)*e^(2*r) - e^(2*r)*d(f)/dr 
E_y2,y2 = -2*(nu + 1)*e^(2*r)*f(r) - 1/12*(mu*q^2 - 4*Lamb)*e^(2*r) - e^(2*r)*d(f)/dr 
E_r,r = 1/24*(lamb^2*mu*q^2 + 8*Lamb*lamb^2 - 12*lamb^2*d^2(f)/dr^2 - 48*(lamb^2*nu^2 + lamb^2 + 4)*f(r) - 24*(2*lamb^2*nu + lamb^2)*d(f)/dr)/(lamb^2*f(r))

We note that `EE==0` leads to only 4 independent equations:

In [29]:
eq0 = EE[0,0]/exp(2*nu*r)
eq0

2*(nu^2 + nu)*f(r)^2 + (2*nu + 1)*f(r)*d(f)/dr - 1/24*(mu*q^2 + 8*Lamb)*f(r) + 1/2*f(r)*d^2(f)/dr^2

In [30]:
eq1 = EE[1,1]/exp(2*nu*r)
eq1

1/24*mu*q^2 - 2*(nu^2 + nu)*f(r) - nu*d(f)/dr + 1/3*Lamb

In [31]:
eq2 = EE[2,2]/exp(2*r)
eq2

-1/12*mu*q^2 - 2*(nu + 1)*f(r) + 1/3*Lamb - d(f)/dr

In [32]:
eq3 = EE[4,4]*lamb^2*f(r)
eq3

1/24*lamb^2*mu*q^2 + 1/3*Lamb*lamb^2 - 1/2*lamb^2*d^2(f)/dr^2 - 2*(lamb^2*nu^2 + lamb^2 + 4)*f(r) - (2*lamb^2*nu + lamb^2)*d(f)/dr

### Dilaton field equation

First we evaluate $\nabla_m \nabla^m \phi$:

In [33]:
nab = g.connection()
print(nab)

Levi-Civita connection nabla_g associated with the Lorentzian metric g on the 5-dimensional differentiable manifold M


In [34]:
box_phi = nab(nab(phi).up(g)).trace()
print(box_phi)
box_phi.display()

Scalar field on the 5-dimensional differentiable manifold M


M --> R
(t, x, y1, y2, r) |--> 4*(2*(nu + 1)*f(r) + d(f)/dr)/lamb

From the action (1), the field equation for $\phi$ is 
$$ \nabla_m \nabla^m \phi = \frac{\lambda}{4} e^{\lambda\phi} F_{mn} F^{mn}$$
We write it as

    DE == 0
    
with `DE` defined by

In [35]:
DE = box_phi - lamb/4*exp(lamb*phi) * F2
print(DE)
DE.display()

Scalar field on the 5-dimensional differentiable manifold M


M --> R
(t, x, y1, y2, r) |--> -1/8*(lamb^2*mu*q^2 - 64*(nu + 1)*f(r) - 32*d(f)/dr)/lamb

Hence the dilaton field equation provides a fourth equation:

In [36]:
eq4 = DE.coord_function()*lamb
eq4

-1/8*lamb^2*mu*q^2 + 8*(nu + 1)*f(r) + 4*d(f)/dr

### Maxwell equation

From the action (1), the field equation for $F$ is 
$$ \nabla_m \left( e^{\lambda\phi} F^{mn} \right)= 0 $$
We write it as

    ME == 0
    
with `ME` defined by

In [37]:
ME = nab(exp(lamb*phi)*Fu).trace(0,2)
print(ME)
ME.display()

Vector field on the 5-dimensional differentiable manifold M


0

We get identically zero; indeed the tensor $\nabla_p (e^{\lambda\phi} F^{mn})$ has a vanishing trace, as we can check:

In [38]:
nab(exp(lamb*phi)*Fu).display()

mu*q d/dy1*d/dy2*dr - 1/2*mu*q*e^(2*r)*f(r) d/dy1*d/dr*dy2 - mu*q d/dy2*d/dy1*dr + 1/2*mu*q*e^(2*r)*f(r) d/dy2*d/dr*dy1 + 1/2*mu*q*e^(2*r)*f(r) d/dr*d/dy1*dy2 - 1/2*mu*q*e^(2*r)*f(r) d/dr*d/dy2*dy1

### Solving the field equations

The system to solve is

In [39]:
eqs = [eq0, eq1, eq2, eq3, eq4]
for eq in eqs:
    pretty_print(eq, ' = 0')

2*(nu^2 + nu)*f(r)^2 + (2*nu + 1)*f(r)*d(f)/dr - 1/24*(mu*q^2 + 8*Lamb)*f(r) + 1/2*f(r)*d^2(f)/dr^2 ' = 0'

1/24*mu*q^2 - 2*(nu^2 + nu)*f(r) - nu*d(f)/dr + 1/3*Lamb ' = 0'

-1/12*mu*q^2 - 2*(nu + 1)*f(r) + 1/3*Lamb - d(f)/dr ' = 0'

1/24*lamb^2*mu*q^2 + 1/3*Lamb*lamb^2 - 1/2*lamb^2*d^2(f)/dr^2 - 2*(lamb^2*nu^2 + lamb^2 + 4)*f(r) - (2*lamb^2*nu + lamb^2)*d(f)/dr ' = 0'

-1/8*lamb^2*mu*q^2 + 8*(nu + 1)*f(r) + 4*d(f)/dr ' = 0'

Let us solve `eq1` for $f(r)$:

In [40]:
sol_f = desolve(eq1.expr() == 0, f(r), ivar=r)
sol_f.expand()

_C*e^(-2*(nu + 1)*r) + 1/48*mu*q^2/((nu + 1)*nu) + 1/6*Lamb/((nu + 1)*nu)

Hence, up to some rescaling the solution is of the type

$$ f(r) = 1 - m e^{-(2\nu +2)r}, $$

where $m$ is a constant. Hence we declare

In [41]:
var('m', domain='real')
fm(r) = 1 - m*exp(-(2*nu+2)*r)
fm

r |--> -m*e^(-2*(nu + 1)*r) + 1

and substitute this function for $f(r)$ in all the equations:

In [42]:
eq0m = eq0.expr().substitute_function(f, fm).simplify_full()
eq0m

1/24*(m*mu*q^2 - 48*m*nu^2 + 8*Lamb*m - 48*m*nu - (mu*q^2 - 48*nu^2 + 8*Lamb - 48*nu)*e^(2*nu*r + 2*r))*e^(-2*nu*r - 2*r)

In [43]:
eq0m = (eq0m * exp(2*nu*r+2*r)).simplify_full()
eq0m

1/24*m*mu*q^2 - 2*m*nu^2 + 1/3*Lamb*m - 2*m*nu - 1/24*(mu*q^2 - 48*nu^2 + 8*Lamb - 48*nu)*e^(2*nu*r + 2*r)

In [44]:
eq1m = eq1.expr().substitute_function(f, fm).simplify_full()
eq1m

1/24*mu*q^2 - 2*nu^2 + 1/3*Lamb - 2*nu

In [45]:
eq2m = eq2.expr().substitute_function(f, fm).simplify_full()
eq2m

-1/12*mu*q^2 + 1/3*Lamb - 2*nu - 2

In [46]:
eq3m = eq3.expr().substitute_function(f, fm).simplify_full()
eq3m

-1/24*(48*lamb^2*m*nu - 48*(lamb^2 + 4)*m - (lamb^2*mu*q^2 - 48*lamb^2*nu^2 + 8*(Lamb - 6)*lamb^2 - 192)*e^(2*nu*r + 2*r))*e^(-2*nu*r - 2*r)

In [47]:
eq3m = (eq3m * exp(2*nu*r+2*r)).simplify_full()
eq3m

-2*lamb^2*m*nu + 2*(lamb^2 + 4)*m + 1/24*(lamb^2*mu*q^2 - 48*lamb^2*nu^2 + 8*(Lamb - 6)*lamb^2 - 192)*e^(2*nu*r + 2*r)

In [48]:
eq4m = eq4.expr().substitute_function(f, fm).simplify_full()
eq4m

-1/8*lamb^2*mu*q^2 + 8*nu + 8

In [49]:
eqs = [eq0m, eq1m, eq2m, eq3m, eq4m]

### Solution for $\nu = 2$

In [50]:
neqs = [eq.subs(nu=2).simplify_full() for eq in eqs]
[eq == 0 for eq in neqs]

[1/24*m*mu*q^2 + 1/3*(Lamb - 36)*m - 1/24*(mu*q^2 + 8*Lamb - 288)*e^(6*r) == 0,
 1/24*mu*q^2 + 1/3*Lamb - 12 == 0,
 -1/12*mu*q^2 + 1/3*Lamb - 6 == 0,
 -2*(lamb^2 - 4)*m + 1/24*(lamb^2*mu*q^2 + 8*(Lamb - 30)*lamb^2 - 192)*e^(6*r) == 0,
 -1/8*lamb^2*mu*q^2 + 24 == 0]

In [51]:
solve([eq == 0 for eq in neqs], lamb, mu, Lamb, q, m, r)

[[lamb == 2, mu == 48/r1^2, Lamb == 30, q == r1, m == r2, r == r3], [lamb == -2, mu == 48/r4^2, Lamb == 30, q == r4, m == r5, r == r6]]

In the above solutions, $r_i$, with $i$ an integer, stands for an arbitrary parameter. In particular, we notice that $\mu$ and $q$ are related by $\mu q^2 = 48$ and that the value of $m$ can be chosen arbitrarily.

### Solution for $\nu=4$

In [52]:
neqs = [eq.subs(nu=4).simplify_full() for eq in eqs]
[eq == 0 for eq in neqs]

[1/24*m*mu*q^2 + 1/3*(Lamb - 120)*m - 1/24*(mu*q^2 + 8*Lamb - 960)*e^(10*r) == 0,
 1/24*mu*q^2 + 1/3*Lamb - 40 == 0,
 -1/12*mu*q^2 + 1/3*Lamb - 10 == 0,
 -2*(3*lamb^2 - 4)*m + 1/24*(lamb^2*mu*q^2 + 8*(Lamb - 102)*lamb^2 - 192)*e^(10*r) == 0,
 -1/8*lamb^2*mu*q^2 + 40 == 0]

In [53]:
solve([eq == 0 for eq in neqs], lamb, mu, Lamb, q, m, r)

[[lamb == 2/3*sqrt(3), mu == 240/r7^2, Lamb == 90, q == r7, m == r8, r == r9], [lamb == -2/3*sqrt(3), mu == 240/r10^2, Lamb == 90, q == r10, m == r11, r == r12]]

As above, $r_i$, with $i$ an integer, stands for an arbitrary parameter. The constants $\mu$ and $q$ are now related by $\mu q^2 = 240$ and the value of $m$ is still arbitrary.